### PR TITLE
support timezone

### LINKED ISSUE #1688: TimeZone support?
How to support the time zone function? If i don't use the time zone for daylight saving time and Eastern time, it will be very difficult to handle.

### PR DESCRIPTION
**Pull request by sunkai-cai**
Fixes #1688.

Changes proposed in this pull request:
- add `timezone` in jobConfiguration
- set `timezone` when build Trigger of quartz

### CODE REVIEW

**Comment by sunkai-cai** [comment]
support spring timezone in the next PR

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduleController.java [hunk 1]
```diff
@@ -88,8 +91,23 @@ public synchronized void rescheduleJob() {
...
-    private Trigger createCronTrigger(final String cron) {
-        return TriggerBuilder.newTrigger().withIdentity(triggerIdentity).withSchedule(CronScheduleBuilder.cronSchedule(cron).withMisfireHandlingInstructionDoNothing()).build();
+    private Trigger createCronTrigger(final String cron, final String timeZoneString) {
+        return TriggerBuilder.newTrigger().withIdentity(triggerIdentity).withSchedule(
+                CronScheduleBuilder.cronSchedule(cron).inTimeZone(parseTimeZoneString(timeZoneString)).withMisfireHandlingInstructionDoNothing()
+        ).build();
+    }
+    
+    private TimeZone parseTimeZoneString(final String timeZoneString) {
+        TimeZone timeZone;
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduleController.java]: Name the variable `result`, please.

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduleController.java [hunk 2]
```diff
@@ -88,8 +91,23 @@ public synchronized void rescheduleJob() {
...
+        if (null != timeZoneString) {
+            timeZone = TimeZone.getTimeZone(timeZoneString);
+            if ("GMT".equals(timeZone.getID()) && !timeZoneString.startsWith("GMT")) {
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduleController.java]: Consider using `Preconditions`.

File: elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/internal/config/RescheduleListenerManagerTest.java [hunk 3]
```diff
@@ -63,19 +63,19 @@ public void assertStart() {
     @Test
     public void assertCronSettingChangedJobListenerWhenIsNotCronPath() {
         rescheduleListenerManager.new CronSettingAndJobEventChangedJobListener().dataChanged("/test_job/config/other", Type.NODE_CREATED, LiteYamlConstants.getJobYaml());
-        verify(jobScheduleController, times(0)).rescheduleJob(ArgumentMatchers.any());
+        verify(jobScheduleController, times(0)).rescheduleJob(ArgumentMatchers.any(), ArgumentMatchers.any());
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/internal/config/RescheduleListenerManagerTest.java]: Use static import for `ArgumentMatchers`.

**Review by TeslaCN — CHANGES_REQUESTED** [review]

File: elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/internal/config/RescheduleListenerManagerTest.java (hunk 3 above)
Comment (sunkai-cai) [reply elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/internal/config/RescheduleListenerManagerTest.java]: ok

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduleController.java [hunk 4]
```diff
@@ -88,8 +92,25 @@ public synchronized void rescheduleJob() {
...
+     * Get the TimeZone for the time zone specification.
+     *
+     * @param timeZoneString must start with "GMT", such as "GMT+8:00"
+     * @return the specified TimeZone, or the GMT zone if the `timeZoneString` cannot be understood.
+     */
+    private TimeZone parseTimeZoneString(final String timeZoneString) {
+        if (null == timeZoneString) {
+            return TimeZone.getDefault();
+        }
+
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduleController.java]: Please remove the redundant line.

Comment (sunkai-cai) [reply elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduleController.java]: ok

**Review by TeslaCN — APPROVED** [review]

**Pull request merged**